src/util/android/event_loop_signal.hpp

```diff
@@ -0,0 +1,131 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#include <atomic>
+
+#include <errno.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <android/log.h>
+#include <android/looper.h>
+
+#define LOGE(fmt...) do { \
+    fprintf(stderr, fmt); \
+    __android_log_print(ANDROID_LOG_ERROR, "REALM", fmt); \
+} while (0)
+
+namespace realm {
+namespace util {
+template<typename Callback>
+class EventLoopSignal : std::enable_shared_from_this() {
```

Comment:
I think this should be `std::enable_shared_from_this<EventLoopSignal<Callback>>`?